datasafe-directory/datasafe-directory-impl/src/main/java/de/adorsys/datasafe/business/impl/profile/operations/DFSBasedProfileStorageImpl.java

```diff
@@ -85,44 +89,56 @@ public void registerPrivate(CreateUserPrivateProfile profile) {
 
     @Override
     public void deregister(UserIDAuth userID) {
-        ListRequest<UserIDAuth, AbsoluteResourceLocation<PrivateResource>> privateRequest =
-                new ListRequest<>(userID, privateProfile(userID).getPrivateStorage());
+        if (!userExists(userID.getUserID())) {
+            log.debug("User deregistation failed. User '{}' does not exist", userID);
```

Comment:
may be here would be better business exception?

Reply:
or return false

Reply:
I've added exception